numpyro/infer/mcmc.py

```diff
@@ -1250,8 +1252,12 @@ def get_extra_fields(self, group_by_chain=False):
         states = self._states if group_by_chain else self._states_flat
         return {k: v for k, v in states.items() if k != 'z'}
 
-    def print_summary(self, prob=0.9):
-        print_summary(self._states['z'], prob=prob)
+    def print_summary(self, prob=0.9, exclude_deterministic=True):
+        # Exclude deterministic sites by default
+        sites = self._states['z']
+        if exclude_deterministic:
```

Comment:
nit: I think you would need to add a check `if isinstance(sites, dict) and exclude_deterministic:` here in case `potential_fn` is used instead of `model`.

Reply:
Good catch, will fix.